### PR TITLE
Bank Statement Import directly in Bank Statement window

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/metasfresh/issues/6061

### CODE REVIEW

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/bankstatement_allocateInvoicesProcess/BankStatement_AllocateInvoicesService.java  2
- src/main/java/de/metas/banking/process/bankstatement_allocateInvoicesProcess/PaymentsForInvoicesCreator.java  6
- src/test/java/de/metas/banking/process/BankStatementTestHelper.java  1
- src/test/java/de/metas/banking/process/C_BankStatement_AllocateInvoicesTest.java  1
- src/main/java/de/metas/banking/process/C_BankStatement_AllocateInvoices.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

File: src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	private LookupValuesList paymentLookupProvider()
+	{
+		final I_C_BankStatementLine line = getSelectedBankStatementLine();
+		final CurrencyId currencyId = CurrencyId.ofRepoId(line.getC_Currency_ID());
+		final boolean isReceipt = line.getStmtAmt().signum() >= 0;
+		final BigDecimal paymentAmount = isReceipt ? line.getStmtAmt() : line.getStmtAmt().negate();
+
+		final ImmutableSet<PaymentId> paymentIds = Services.get(IPaymentDAO.class).retrieveAllMatchingPayments(isReceipt, paymentAmount, currencyId, bPartnerId);
+
+		final LookupDescriptor paymentByIdLookupDescriptor = SqlLookupDescriptor.builder()
```
Comment (teosarca) [inline src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java]: use `searchInTable`

File: src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+				.setWidgetType(DocumentFieldWidgetType.Lookup)
+				.buildForDefaultScope();
+
+		return LookupDataSourceFactory.instance.getLookupDataSource(paymentByIdLookupDescriptor).findByIdsOrdered(paymentIds);
+	}
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		just___DO_IT(getSelectedBankStatementLine(), paymentId);
```
Comment (teosarca) [inline src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java]: dude....

File: src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java (hunk 1 above)
Comment (TheBestPessimist) [reply src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java]: done

File: src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java (hunk 2 above)
Comment (TheBestPessimist) [reply src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java]: done :(

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatementLine_AllocatePayment.java  8
- src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatementLine_AllocatePayment.java  4
- src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatementLine_AllocatePayment.java  4
- src/main/java/de/metas/banking/process/C_BankStatement_AddBpartnerAndPayment.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatement_AllocatePayment.java  4
- src/main/java/de/metas/banking/process/C_BankStatementLine_AddBpartnerAndPayment.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatementLine_AddBpartnerAndPayment.java  8
- src/main/java/de/metas/banking/process/C_BankStatement_AllocatePayment.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatement_AllocatePayment.java  4
- src/main/java/de/metas/banking/process/C_BankStatementLine_AddBpartnerAndPayment.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/metas/banking/process/C_BankStatementLine_AddBpartnerAndPayment.java  8
- src/main/java/de/metas/banking/process/C_BankStatement_AllocatePayment.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh-webui-api/pullRequest?prid=4898014&bid=16219737)

**Pull request merged**